File: cs/XDust/XDust/XDustSepNode.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace XDust
{
    public class XDustSepNode : XDustNodeList, IXDustSectionNode
    {
        public XDustSepNode(IEnumerable<XDustNode> nodes)
            : base(nodes)
        {
            this.Parameters = new Dictionary<string, XDustNode>();
        }
        
        public Dictionary<String, XDustNode> Parameters { get; private set; }

        public override object PrepareModel(RenderChain chain, Context context, object model)
        {
            IScriptable scriptable = new Scriptable(model);
            Object sep = scriptable["@sep"];
            if (sep != null && (bool)sep)
            {
                return base.PrepareModel(chain, context, model);
            }
            else
            {
                return null;
            }
        }

        public override string ToString()
        {
            return String.Format("{{@sep}}{0}{{/sep}}", base.ToString());
        }

        #region IXDustSectionNode Members


        public XDustNodeList StartBody(string name)
        {
            throw new NotImplementedException();
        }

        public void EndBody()
        {
            throw new NotImplementedException();
        }

        #endregion
    }
}

File: java/src/xdust/src/com/heydanno/xdust/XDustSepNode.java
package com.heydanno.xdust;

import java.lang.reflect.InvocationTargetException;
import java.util.Collection;
import java.util.HashMap;
import java.util.Map;

import sun.reflect.generics.reflectiveObjects.NotImplementedException;

public class XDustSepNode extends XDustNodeList implements IXDustSectionNode {

	public XDustSepNode(Collection<XDustNode> nodes) {
		super(nodes);
	}
	
	private Map<String, XDustNode> parameters;

	@Override
	public Map<String, XDustNode> getParameters() {
		if (null == this.parameters) {
			this.parameters = new HashMap<String, XDustNode>();
		}
		return this.parameters;
	}
	
	@Override
	public Object prepareModel(RenderChain chain, Context context, Object model) {
		IScriptable scriptable = Scriptable.from(model);
		Object sep;
		try {
			sep = scriptable.get("@sep");
		} catch (IllegalArgumentException e) {
			return null;
		} catch (IllegalAccessException e) {
			return null;
		} catch (InvocationTargetException e) {
			return null;
		}
		if (sep != null && (Boolean)sep) {
			return super.prepareModel(chain, context, model);
		} else {
			return null;
		}
	}
	
	@Override
	public String toString() {
		return String.format("{@sep}%s{/sep}", super.toString());
	}

	@Override
	public XDustNodeList startBody(String name) {
		throw new NotImplementedException();
	}

	@Override
	public void endBody() {
		throw new NotImplementedException();
	}

}
